<launch>
    <!-- Subscribed topics -->
    <param name="/LiDAR_pointcloud_topic" type="string" value= "/laser_cloud_flat" />
    <param name="/IMU_topic" type="string" value= "/os_cloud_node/imu" />
    <param name="/Image_topic" type="string" value= "/NotAvail" />
    <param name="map_output_dir" type="string" value="$(env HOME)/lvi_fusion_output" />
    <rosparam command="load" file="$(find lvi_fusion)/../config/lvi_fusion_config.yaml" />
    
    <!-- set LiDAR type as ouster-64 spining LiDAR -->
    <param name="/Lidar_front_end/lidar_type" type="int" value= "3" /> 
    <param name="/Lidar_front_end/point_step" type="int" value="1" />
    <param name="lvi_fusion_lio/lio_update_point_step" type="int" value="6" />
        
    <node pkg="lvi_fusion" type="lvi_fusion_LiDAR_front_end" name="lvi_fusion_LiDAR_front_end"  output="screen" required="true"/>
    <node pkg="lvi_fusion" type="lvi_fusion_mapping" name="lvi_fusion_mapping" output="screen" required="true" />
    
    <arg name="rviz" default="1" />
    <group if="$(arg rviz)">
        <node name="rvizvisualisation" pkg="rviz" type="rviz" output="log" args="-d $(find lvi_fusion)/../config/rviz/lvi_fusion_rviz_config_ouster.rviz" />
    </group>
 </launch>
